File: src/lib/server/validation/responses/day-properties.ts
import 'zod-openapi';
import { z } from 'zod';

const dayPropertiesSchema = z
	.object({
		date: z.string(),
		isFastingDay: z.boolean(),
		notes: z.string().nullable(),
		waterMl: z.number().int().nullable(),
		activityCalories: z.number().int().nullable(),
		activityNote: z.string().nullable()
	})
	.meta({ id: 'DayProperties' });

export const dayPropertiesResponseSchema = z
	.object({
		properties: dayPropertiesSchema.nullable()
	})
	.meta({ id: 'DayPropertiesResponse' });

export const dayPropertiesRangeResponseSchema = z
	.object({
		data: z.array(dayPropertiesSchema)
	})
	.meta({ id: 'DayPropertiesRangeResponse' });

File: mobile/shared/src/commonMain/kotlin/com/bissbilanz/model/DayProperties.kt
package com.bissbilanz.model

import kotlinx.serialization.Serializable

@Serializable
data class DayProperties(
    val date: String,
    val isFastingDay: Boolean,
)

@Serializable
data class DayPropertiesResponse(
    val properties: DayProperties? = null,
)

@Serializable
data class DayPropertiesRangeResponse(
    val data: List<DayProperties>,
)

@Serializable
data class DayPropertiesSet(
    val date: String,
    val isFastingDay: Boolean,
)
